test/test_nn.py

```diff
@@ -13356,6 +13338,77 @@ def test_grid_sample_nan_inf(self, device, dtype):
                                                      padding_mode=padding_mode, align_corners=False)
             self.assertEqual(sample, torch.zeros([1, 1, 1, 2], device=device, dtype=dtype))
 
+    @onlyOnCPUAndCUDA
+    def test_fractional_max_pool2d(self, device):
+        x = torch.randn(1, 2, 7, 7, requires_grad=True, device=device)
+        samples = x.new(1, 2, 2).uniform_()
+
+        def func(x):
+            return F.fractional_max_pool2d(
+                x, (2, 2), output_size=(3, 3), _random_samples=samples)
+
+        self.assertEqual(func(x).shape, (1, 2, 3, 3))
+        gradcheck(func, [x])
+        gradgradcheck(func, [x])
+
+        if self.device_type != 'cuda':
```

Comment:
Why can't this run on cuda?

Reply:
https://github.com/pytorch/pytorch/issues/51622

```python
import torch
x = torch.randn(2, 7, 7, requires_grad=True, device='cpu')
samples = x.new(2, 2).uniform_()
torch.nn.functional.fractional_max_pool2d(x, (2, 2), output_size=(3, 3), _random_samples=samples)
tensor([[[ 0.7063,  0.2786,  0.8529],
         [ 0.7309,  0.8398,  1.5179],
         [ 0.5881,  1.4823,  0.6529]],

        [[-0.1289,  1.4571,  0.5930],
         [ 0.6710,  0.3553,  0.9126],
         [ 0.6864,  0.3350,  0.6807]]], grad_fn=<FractionalMaxPool2DBackward>)
torch.nn.functional.fractional_max_pool2d(x.cuda(), (2, 2), output_size=(3, 3), _random_samples=samples.cuda())
RuntimeError: TensorAccessor expected 3 dims but tensor has 2
```

It fails on CUDA. The same thing works on CPU and was part of the original test (which only checked for CPU). Moving it to CUDA caused this error.
Forgot to add link to the issue.

Thanks!

Reply:
You can change x.new(2, 2)... to x.new(1, 2, 2)... that should work on both CPU and CUDA.